src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceIndexingOperationListenerImpl.java

```diff
@@ -121,49 +128,51 @@ public Index preIndex(final ShardId shardId, final Index index) {
     @Override
     public void postIndex(final ShardId shardId, final Index index, final Exception ex) {
         final ComplianceConfig complianceConfig = auditlog.getComplianceConfig();
-        if (complianceConfig.isEnabled() && complianceConfig.shouldLogDiffsForWrite()) {
+        if (isComplianceEnabledForIndex(complianceConfig) && complianceConfig.shouldLogDiffsForWrite()) {
             threadContext.remove();
         }
     }
 
     @Override
     public void postIndex(ShardId shardId, Index index, IndexResult result) {
         final ComplianceConfig complianceConfig = auditlog.getComplianceConfig();
-        if (complianceConfig.isEnabled() && complianceConfig.shouldLogDiffsForWrite()) {
-            final Context context = threadContext.get();
-            final GetResult previousContent = context==null?null:context.getGetResult();
-            threadContext.remove();
-            Objects.requireNonNull(is);
-
-            if (result.getFailure() != null || index.origin() != org.elasticsearch.index.engine.Engine.Operation.Origin.PRIMARY) {
-                return;
-            }
+        if (isComplianceEnabledForIndex(complianceConfig)) {
+            if (complianceConfig.shouldLogDiffsForWrite()) {
+                final Context context = threadContext.get();
+                final GetResult previousContent = context == null ? null : context.getGetResult();
+                threadContext.remove();
+                Objects.requireNonNull(is);
+
+                if (result.getFailure() != null || index.origin() != org.elasticsearch.index.engine.Engine.Operation.Origin.PRIMARY) {
+                    return;
+                }
 
-            if(is.getShardOrNull(shardId.getId()) == null) {
-                return;
-            }
+                if (is.getShardOrNull(shardId.getId()) == null) {
```

Comment:
You can club this with the if statement on the line above https://github.com/opendistro-for-elasticsearch/security/pull/475/files#diff-5c8a4dc5ef259d5edbdab7104f8ff0f2R146